src/daft-dsl/src/expr/mod.rs

```diff
@@ -990,21 +990,9 @@ impl Expr {
                     to_sql_inner(inner, buffer)?;
                     write!(buffer, ") IS NOT NULL")
                 }
-                Expr::IfElse {
```

Comment:
SQL server only accepts values in `case when then` clauses, so it was failing on the `if_else` pushdowns test,  because we test with conditional expressions inside the `if_else` branches. 

I think it's better to just remove it, the benefit of having if_else pushdowns in read_sql is probably not much at all, and i'd instead rather just avoid having this be a bug in the future.